eventbus/kafka/sensor/kafka_sensor.go

```diff
@@ -0,0 +1,367 @@
+package kafka
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Knetic/govaluate"
+	"github.com/Shopify/sarama"
+	eventbuscommon "github.com/argoproj/argo-events/eventbus/common"
+	"github.com/argoproj/argo-events/eventbus/kafka/base"
+	eventbusv1alpha1 "github.com/argoproj/argo-events/pkg/apis/eventbus/v1alpha1"
+	sensorv1alpha1 "github.com/argoproj/argo-events/pkg/apis/sensor/v1alpha1"
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"go.uber.org/zap"
+)
+
+type KafkaSensor struct {
+	*base.Kafka
+	*sync.Mutex
+	sensor *sensorv1alpha1.Sensor
+
+	// kafka
+	topics    *Topics
+	client    sarama.Client
+	consumer  sarama.ConsumerGroup
+	hostname  string
+	groupName string
+
+	// triggers
```

Comment:
can we expand on this comment? looks like it's actual kafka connections?

Reply:
The triggers is a map of KafkaTriggerHandler (specifically `map[string]KafkaTriggerHandler`), which is an implementation of the TriggerHandler interface. With Jetstream and Nats, this information is not required in the sensor because events are consumed by individual triggers. However, with kafka we need this information here in the Sensor because events are consumed centrally. Will expand in the comment :+1: